game/src/sandbox/gameplay/mod.rs

```diff
@@ -118,11 +134,27 @@ impl GameplayMode {
         } else if name == "home_to_work" {
             LoadScenario::Scenario(ScenarioGenerator::proletariat_robot(map, &mut rng, timer))
         } else if name == "census" {
-            let config = popdat::Config::default();
-            LoadScenario::Scenario(
-                popdat::generate_scenario("typical monday", config, map, &mut rng)
-                    .expect("unable to build census scenario"),
-            )
+            let map_area = map.get_boundary_polygon().clone();
+            let map_bounds = map.get_gps_bounds().clone();
+            let mut rng = sim::fork_rng(&mut rng);
+
+            LoadScenario::Future(Box::pin(async move {
+                let areas = popdat::CensusArea::fetch_all_for_map(&map_area, &map_bounds).await?;
+
+                let scenario_from_app: Box<dyn Send + FnOnce(&App) -> Scenario> =
```

Comment:
That's right. I'd like to move more of the computation onto the background thread (at least for native) but right now it all requires Map which is not Send.

Reply:
Hmm, that might be easy to fix. I think the only reason it's not `Send` is because of the `path_calc: ThreadLocal<RefCell<PathCalculator>>` trick deep inside the pathfinding layer. It may not be hard to rearchitect and plumb the `PathCalculator` in. (It's an internal `fast_paths` thing that makes it cheaper to run queries repeatedly on the same thread.)